sdk/cosmosdb/cosmos/src/client/SasToken/SasToken.ts

```diff
@@ -0,0 +1,307 @@
+    } else if (typeof keyType === "object") {
+      switch (keyType) {
+        case CosmosKeyType.PrimaryMaster:
+          this.keyType = 1;
+        case CosmosKeyType.SecondaryMaster:
+          this.keyType = 2;
+          break;
+        case CosmosKeyType.PrimaryReadOnly:
+          this.keyType = 3;
+          break;
+        case CosmosKeyType.SecondaryReadOnly:
+          this.keyType = 4;
+          break;
+        default:
+          throw new Error("keyType");
+          break;
+      }
+      return this.SasTokenWithHMACSHA256(key);
+    } else {
+      throw new Error("key");
+    }
+  }
+
+  public get DatabaseName(): string {
+    return this.databaseName;
+  }
+
+  private generatePayload(): string {
+    let resourcePrefixPath: string = "";
+    if (!(this.databaseName === "")) {
+      resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.DatabasesPathSegment}/${this.databaseName}`;
+    }
+
+    if (!(this.containerName === "")) {
+      if (this.databaseName === "") {
+        throw new Error("databaseName");
+      }
+      resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.CollectionsPathSegment}${Constants.Path.Root}${this.containerName}`;
+    }
+
+    if (!(this.resourceName === "")) {
+      if (this.containerName === "") {
+        throw new Error("containerName");
+      }
+
+      switch (this.resourceKind) {
+        case CosmosContainerChildResourceKind.Item:
+          resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.DocumentsPathSegment}`;
+          break;
+        case CosmosContainerChildResourceKind.StoredProcedure:
+          resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.StoredProceduresPathSegment}`;
+          break;
+        case CosmosContainerChildResourceKind.UserDefinedFunction:
+          resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.UserDefinedFunctionsPathSegment}`;
+          break;
+        case CosmosContainerChildResourceKind.Trigger:
+          resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${Constants.Path.TriggersPathSegment}`;
+          break;
+        default:
+          throw new Error("resourceKind");
+          break;
+      }
+
+      resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}${this.resourceName}`;
+    }
+
+    resourcePrefixPath = `${resourcePrefixPath}${Constants.Path.Root}`;
+    this.resourcePath = resourcePrefixPath.toString();
+    if (this.partitionKeyValueRanges != null && !(this.partitionKeyValueRanges.length > 0)) {
+      if (this.resourceKind !== CosmosContainerChildResourceKind.Item) {
+        throw new Error("partitionKeyValueRanges");
+      }
+
+      for (const range of this.partitionKeyValueRanges) {
+        this.partitionRanges = `${this.partitionRanges}${range.encode()},`;
+      }
+    }
+
+    if (this.expiryTime == null) {
```

Comment:
```suggestion
    if (!this.expiryTime) {
```

We almost always never used explicit `null` by choice